play-json/shared/src/main/scala/play/api/libs/json/JsResult.scala

```diff
@@ -71,6 +73,8 @@ case class JsError(errors: collection.Seq[(JsPath, collection.Seq[JsonValidation
 
   def exists(p: Nothing => Boolean): Boolean = false
 
+  def forall(p: Nothing => Boolean): Boolean = true
+
```

Comment:
I guess we must have it returning `true` just to align with other `forall` methods in Scala. However, this is something that always annoyed me. 

`forall` on the empty or error side always returns true in other libraries.

```scala
@ val optEmpty: Option[String] = None
optEmpty: Option[String] = None

@ optEmpty.forall(_.startsWith("a"))
res6: Boolean = true

@ optEmpty.forall(_.startsWith("b"))
res6: Boolean = true
```

In the above example, how does it even make any sense? If the option is empty, all predicates making assertions about its content should return false. The same is valid for Scala collections. 

We will have the same situation here.

Reply:
I asked about it in the Scala forum the motivations for this behaviour. 
https://users.scala-lang.org/t/forall-semantics/3303

Not that I believe we should change it. We should keep it consistent with other libraries.

Reply:
@renatocaval I realized that `_.forall` behavior is pretty lear after reading this blogpost: http://blog.originate.com/blog/2014/06/15/idiomatic-scala-your-options-do-not-match/

Specifically:
![image](https://user-images.githubusercontent.com/11317222/46000721-07a89300-c0b2-11e8-9911-70697552662c.png)
![image](https://user-images.githubusercontent.com/11317222/46000728-0e370a80-c0b2-11e8-8129-7d17efe24995.png)

>It makes perfect sense in plain English: to say “all numbers are prime” is equivalent to saying “there does not exist a number that is not prime”.

Reply:
Thanks @aaabramov. 
I suspected that there was a solid reason for that. TIL